### PR TITLE
Couple projector countdown to speaker speak/end_speech

### PR DESCRIPTION
**Pull request by reiterl**

### CODE REVIEW

File: openslides_backend/action/actions/speaker/end_speech.py [hunk 1]
```diff
@@ -26,12 +27,43 @@ def get_updated_instances(self, action_data: ActionData) -> ActionData:
...
+                    FullQualifiedId(Collection("meeting"), speaker["meeting_id"]),
+                    [
+                        "list_of_speakers_couple_countdown",
+                        "list_of_speakers_countdown_id",
+                    ],
+                )
+                if meeting.get("list_of_speakers_couple_countdown") and meeting.get(
+                    "list_of_speakers_countdown_id"
+                ):
+                    countdown = self.datastore.get(
```
Comment (FinnStutzenstein) [inline openslides_backend/action/actions/speaker/end_speech.py:51]: Same as in #561 : Please unify this, so countdown logic is not present here.

File: openslides_backend/action/actions/speaker/end_speech.py [hunk 2]
```diff
@@ -26,12 +27,43 @@ def get_updated_instances(self, action_data: ActionData) -> ActionData:
...
                 lock_result=True,
             )
             if speaker.get("begin_time") is None or speaker.get("end_time") is not None:
                 raise ActionException(
                     f"Speaker {instance['id']} is not speaking at the moment."
                 )
             instance["end_time"] = round(time.time())
+
+            # reset projector_countdown
+            if speaker.get("meeting_id"):
```
Comment (jsangmeister) [inline openslides_backend/action/actions/speaker/end_speech.py:40]: Also same as in #561: the speaker must have a meeting id.

File: openslides_backend/action/actions/speaker/speak.py [hunk 3]
```diff
@@ -64,4 +65,35 @@ def get_updated_instances(self, action_data: ActionData) -> ActionData:
                         "id": speaker_id,
                         "end_time": now,
                     }
+
+            # reset projector countdown
+            if this_speaker.get("meeting_id"):
```
Comment (jsangmeister) [inline openslides_backend/action/actions/speaker/speak.py:70]: same here

File: openslides_backend/action/actions/speaker/speak.py [hunk 4]
```diff
@@ -64,4 +65,35 @@ def get_updated_instances(self, action_data: ActionData) -> ActionData:
...
+                    FullQualifiedId(Collection("meeting"), this_speaker["meeting_id"]),
+                    [
+                        "list_of_speakers_couple_countdown",
+                        "list_of_speakers_countdown_id",
+                    ],
+                )
+                if meeting.get("list_of_speakers_couple_countdown") and meeting.get(
+                    "list_of_speakers_countdown_id"
+                ):
+                    countdown = self.datastore.get(
```
Comment (jsangmeister) [inline openslides_backend/action/actions/speaker/speak.py:81]: same here

**Review by jsangmeister — CHANGES_REQUESTED** [review]

File: openslides_backend/action/actions/speaker/end_speech.py (hunk 1 above)
Comment (reiterl) [reply openslides_backend/action/actions/speaker/end_speech.py:51]: Unified.

File: openslides_backend/action/actions/speaker/end_speech.py (hunk 2 above)
Comment (reiterl) [reply openslides_backend/action/actions/speaker/end_speech.py:40]: Removed this check.

File: openslides_backend/action/actions/speaker/speak.py (hunk 3 above)
Comment (reiterl) [reply openslides_backend/action/actions/speaker/speak.py:70]: removed this check.

File: openslides_backend/action/actions/speaker/speak.py (hunk 4 above)
Comment (reiterl) [reply openslides_backend/action/actions/speaker/speak.py:81]: Unified.

**Review by jsangmeister — APPROVED** [review]

**Pull request merged**